Description: Distinguish ssize_t error code from size_t value
 'writev' returns ssize_t value, which could contain error code (-1) or
 number of written bytes.
 .
 When returned value assigned to size_t variable and checked for error code (in
 one shot), GCC warns about 'comparison of unsigned expression < 0 is always
 false'.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Forwarded: http://lists.unbit.it/pipermail/uwsgi/2010-September/000666.html
Last-Update: 2010-09-26

Index: uwsgi-0.9.6.4/plugins/rack/rack_plugin.c
===================================================================
--- uwsgi-0.9.6.4.orig/plugins/rack/rack_plugin.c	2010-09-26 07:09:00.000000000 +0300
+++ uwsgi-0.9.6.4/plugins/rack/rack_plugin.c	2010-09-26 09:16:53.000000000 +0300
@@ -331,7 +331,7 @@
         	wsgi_req->hvec[4].iov_base = "\r\n";
         	wsgi_req->hvec[4].iov_len = 2 ;
 
-		if ( (wsgi_req->response_size = writev(wsgi_req->poll.fd, wsgi_req->hvec, 5)) < 0) {
+		if ( !(wsgi_req->response_size = writev(wsgi_req->poll.fd, wsgi_req->hvec, 5)) ) {
                 	uwsgi_error("writev()");
         	}
 
